lib/pages/task_list.dart

```diff
@@ -17,90 +19,69 @@ class _TaskListState extends State<TaskList> {
     context.read<TaskCubit>().readTasks();
     return BlocBuilder<TaskCubit, TaskState>(
       builder: (context, state) {
-        List<Widget> tasks = state.taskNames.map(
-          (taskName) {
+        List<Widget> tasks = state.tasks.map(
+          (task) {
             return ListTile(
-              leading: Icon(
-                Icons.task,
-                color: iconColor,
+              key: const Key('task tiles'),
+              leading: GestureDetector(
+                onTap: () => _showEditPanel(context, task),
+                child: Icon(
+                  Icons.edit,
+                  color: iconColor,
+                ),
               ),
               title: Text(
-                taskName.name,
+                task.name,
                 style: tilesText,
               ),
               trailing: GestureDetector(
-                  onTap: () {
-                    context.read<TaskCubit>().completeTask(taskName: taskName);
-                    context.read<TaskCubit>().removeTask(taskName: taskName);
-                  },
-                  child: Icon(Icons.delete, color: iconColor)),
+                  onTap: () => _onTapCompleteIcon(task),
+                  child: Icon(Icons.remove_circle_outline, color: iconColor)),
             );
           },
         ).toList();
         return Scaffold(
+          bottomNavigationBar: const BottomNav(),
```

Comment:
You might want to consider in the future moving the Scaffold up a level to the `main.dart` or whatever so that there only needs to be one `BottomNav` instance in the app. For now, please resolve this comment.

Reply:
I think you said we are going to do this together later.